File: client/src/classes/EventMessage.ts
import User from './User';

export default class EventMessage {
  id?: string;
  message = '';
  createdOn = new Date();
  user = new User();
  userId?: string;
  eventId?: string;

  constructor(i?: EventMessage) {
    PF.H.Classes.BuildClass(this, i);
  }

  static Create(eventId?: string, userId?: string, message?: string): EventMessage {
    const date = new Date();
    date.setTime(date.getTime() + 3 * 60 * 60 * 1000);
    return new EventMessage({ eventId: eventId, userId: userId, message: message?.trim(), createdOn: date } as EventMessage);
  }
  static GetClassName(): string {
    return 'eventMessage';
  }
}

File: server/models/EventMessage.go
package models

import (
	"time"

	"github.com/google/uuid"
	"github.com/uptrace/bun"
)

type EventMessage struct {
	bun.BaseModel `bun:"event_messages,alias:event_messages"`
	ID            uuid.NullUUID `bun:"id,pk,type:uuid,default:uuid_generate_v4()" json:"id" `
	Message       string        `json:"message"`
	CreatedOn     time.Time     `bun:",nullzero,notnull" json:"createdOn"`
	UserID        *string       `json:"userId"`
	User          *User         `bun:"rel:belongs-to" json:"user"`
	EventID       *string       `json:"eventId"`
}

type EventMessages []EventMessage
type EventMessagesWithCount struct {
	EventMessages EventMessages `json:"items"`
	Count         int           `json:"count"`
}
